zbus/src/object_server/mod.rs

```diff
@@ -767,6 +767,13 @@ impl ObjectServer {
             // error, or deliver the message as though it had an arbitrary one of those
             // interfaces.
             .ok_or_else(|| fdo::Error::Failed("Missing interface".into()))?;
+        // Check that the message has a member before spawning.
+        // Note that an unknown member will still spawn a task. We should instead gather
+        // all the details for the call before spawning.
+        // See also https://github.com/dbus2/zbus/issues/674 for future of Interface.
```

Comment:
* `FIXME: ` prefix would be nice.
* Feel free to make some changes already here, if needed ([I already made it clear that `Interface` is unstable API](https://github.com/dbus2/zbus/pull/711)). However I don't really think we need to change `Interface` here to be able to check everything upfront. The only/main part we really want to spawn off, is the user-provided handlers.
* This commit is only adding a check using the `hdr` we already have at hand so not sure `💩` is justified. :)

Reply:
💩 : because we are doing multiple time the same check and the check is very limited: you can still spawn task if you provide a Member, even if it doesn't exist on the interface. Iow, I disagree this check is worthwhile.

Reply:
> It would be a major refactoring to move to the proposal I made.

Right, I didn't realize you mean you need **all** of your proposed changes here.

> The gain is great though, because there is no need to lock the interface to get those details: callback, spawn, introspection details etc.

No need to (even read) lock the interface to call method on it? How so? :thinking: Keep in mind that the trait can't be object safe if has associated constants or methods w/o receivers IIRC.

> because we are doing multiple time the same check and the check is very limited

We shouldn't need to check multiple times. Once it's established the the member is present, later code should just assume that it is and `unwrap()`.

> Iow, I disagree this check is worthwhile.

I explained already why I want it but more importantly, the check is already done in the existing code so spawning for messages w/o member present, would be a regression.

Reply:
squashed the 💩